src/ophys_etl/modules/segmentation/graph_utils/feature_vector_rois.py

```diff
@@ -0,0 +1,522 @@
+
+        if pixel_ignore is None or not pixel_ignore[p[0], p[1]]:
+            possible_seeds.append(ii)
+
+    n_seeds = 10
+    if rng is None:
+        rng = np.random.RandomState(87123)
+    chosen = set()
+    chosen.add(seed_pt)
+
+    if len(possible_seeds) > n_seeds:
+        chosen_dex = rng.choice(possible_seeds,
+                                size=n_seeds, replace=False)
+        for ii in chosen_dex:
+            p = np.unravel_index(ii, sub_video.shape[1:])
+            chosen.add(p)
+    else:
+        for ii in possible_seeds:
+            p = np.unravel_index(ii, sub_video.shape[1:])
+            chosen.add(p)
+
+    for chosen_pixel in chosen:
+        trace = sub_video[:, chosen_pixel[0], chosen_pixel[1]]
+        thresh = np.quantile(trace, discard)
+        mask = np.where(trace >= thresh)[0]
+        global_mask.append(mask)
+    global_mask = np.unique(np.concatenate(global_mask))
+
+    # apply timestep mask
+    sub_video = sub_video[global_mask, :, :]
+    shape = sub_video.shape
+    n_pixels = shape[1]*shape[2]
+    n_time = shape[0]
+    traces = sub_video.reshape(n_time, n_pixels).astype(float)
+    del sub_video
+
+    # mask pixels that need to be masked
+    if pixel_ignore is not None:
+        traces = traces[:, np.logical_not(pixel_ignore.flatten())]
+        n_pixels = np.logical_not(pixel_ignore).sum()
+
+    # calculate the Pearson correlation coefficient between pixels
+    mu = np.mean(traces, axis=0)
+    traces -= mu
+    var = np.mean(traces**2, axis=0)
+    traces = traces.transpose()
+
+    pearson = np.ones((n_pixels,
+                       n_pixels),
+                      dtype=float)
+
+    numerators = np.tensordot(traces, traces, axes=(1, 1))/n_time
+
+    for ii in range(n_pixels):
+        local_numerators = numerators[ii, ii+1:]
+        denominators = np.sqrt(var[ii]*var[ii+1:])
+        p = local_numerators/denominators
+        pearson[ii, ii+1:] = p
+        pearson[ii+1:, ii] = p
+
+    features = np.copy(pearson)
+
+    # subtract off minimum of each feature
+    pearson_mins = np.min(pearson, axis=0)
+    for ii in range(n_pixels):
+        features[:, ii] = features[:, ii]-pearson_mins[ii]
+
+    # Normalize by the interquartile range of each feature.
+    # If the interquartile range is 0, normalize by max-min.
+    # If that is also zero, set the norm to 1.0
+    p75 = np.quantile(features, 0.75, axis=0)
+    p25 = np.quantile(features, 0.25, axis=0)
+    pmax = np.max(features, axis=0)
+    pmin = np.min(features, axis=0)
+
+    feature_norms = p75-p25
+    feature_norms = np.where(feature_norms > 1.0e-20,
+                             feature_norms,
+                             pmax-pmin)
+
+    feature_norms = np.where(feature_norms > 1.0e-20,
+                             feature_norms,
+                             1.0)
+
+    for ii in range(n_pixels):
+        features[:, ii] = features[:, ii]/feature_norms[ii]
+
+    return features
```

Comment:
general comment about this function:
it is pretty long. It would probably clarify later if things were parted out, something like:
* `determine_time_mask()`
* `calculate_correlations()`
* `construct_feature_vectors()`

no change requested right away

Reply:
I changed my mind, I think this will help us going forward to split this up

Reply:
Done